warn/scrapers/id.py

```diff
@@ -0,0 +1,127 @@
+import logging
+import random
+import re
+from pathlib import Path
+
+import pdfplumber
+import requests
+
+from .. import utils
+
+__authors__ = ["chriszs"]
+__tags__ = ["pdf"]
+
+logger = logging.getLogger(__name__)
+
+
+def scrape(
+    data_dir: Path = utils.WARN_DATA_DIR,
+    cache_dir: Path = utils.WARN_CACHE_DIR,
+) -> Path:
+    """
+    Scrape data from Idaho.
+
+    Keyword arguments:
+    data_dir -- the Path were the result will be saved (default WARN_DATA_DIR)
+    cache_dir -- the Path where results can be cached (default WARN_CACHE_DIR)
+
+    Returns: the Path where the file is written
+    """
+    state_code = "id"
+    base_url = "https://www.labor.idaho.gov/dnn/Portals/0/Publications/"
+    file_name = "WARNNotice.pdf"
+
+    # There's a numeric parameter called v on this PDF URL that updates
+    # from time to time. Suspect this is a cache-buster. We're using a
+    # random number instead.
+    min_cache_buster = 0
+    max_cache_buster = 10000000000
+    cache_buster = random.randrange(min_cache_buster, max_cache_buster)
+
+    url = f"{base_url}{file_name}?v={cache_buster}"
+
+    cache_state = Path(cache_dir, state_code)
+    cache_state.mkdir(parents=True, exist_ok=True)
+
+    cache_key = f"{cache_state}/{file_name}"
+
+    # verify=False because there's a persistent cert error
+    # we're working around.
+    response = requests.get(url, verify=False)
+    with open(cache_key, "wb") as file:
+        file.write(response.content)
+
+    output_rows: list = []
+
+    with pdfplumber.open(cache_key) as pdf:
+        for index, page in enumerate(pdf.pages):
+            rows = page.extract_table()
+
+            output_rows = output_rows + _clean_table(rows, index)
```

Comment:
```suggestion
            output_rows +=  _clean_table(rows, index)
```
I've actually never written a line like this, but would it work?!